pkg/controller/acmechallenges/finalizer.go

```diff
@@ -32,16 +32,15 @@ import (
 // finalizerRequired returns true if the finalizer is not found on the challenge.
 //
 // API transition
-// We currently only add cmacme.ACMELegacyFinalizer, but a future version will add
-// cmacme.ACMEDomainQualifiedFinalizer.
+// We only add cmacme.ACMEDomainQualifiedFinalizer.
 // A finalizer only needs to be added if neither is present.
 func finalizerRequired(ch *cmacme.Challenge) bool {
 	finalizers := sets.NewString(ch.Finalizers...)
-	return !finalizers.Has(cmacme.ACMELegacyFinalizer) &&
-		!finalizers.Has(cmacme.ACMEDomainQualifiedFinalizer)
+	return !finalizers.Has(cmacme.ACMEDomainQualifiedFinalizer) &&
+		!finalizers.Has(cmacme.ACMELegacyFinalizer)
 }
 
 func otherFinalizerPresent(ch *cmacme.Challenge) bool {
-	return ch.Finalizers[0] != cmacme.ACMELegacyFinalizer &&
-		ch.Finalizers[0] != cmacme.ACMEDomainQualifiedFinalizer
+	return ch.Finalizers[0] != cmacme.ACMEDomainQualifiedFinalizer &&
+		ch.Finalizers[0] != cmacme.ACMELegacyFinalizer
```

Comment:
```suggestion
	finalizers := sets.NewString(ch.Finalizers...)
	finalizers.Delete(cmacme.ACMEDomainQualifiedFinalizer, cmacme.ACMELegacyFinalizer)
	return len(finalizers) > 0
```

Since we are using set-based logic here. Why did we decide just to check the first finalizer? Are we sure our finalizer will come first?

Reply:
The short answer is that it always struck me as odd:
https://github.com/cert-manager/cert-manager/blob/6564f1c3dc81da510c8c5d7d4085100daa2d774b/pkg/controller/acmechallenges/sync.go#L259
and I just faithfully transitioned:
https://github.com/cert-manager/cert-manager/blob/0afe4ab884dcf7b88bfda4e830c487e752675cfa/pkg/controller/acmechallenges/sync.go#L266

Reply:
Thanks! Appears like something that could eventually, though unlikely, fail. Didn't you like my suggestion? 🤣

Reply:
Resolved by https://github.com/cert-manager/cert-manager/pull/7735/files#r2081766963

Reply:
This exact problem is mentioned in https://github.com/cert-manager/cert-manager/pull/5126 and https://github.com/cert-manager/cert-manager/pull/7286, which never got merged: 
* https://github.com/cert-manager/cert-manager/pull/7286